### PR TITLE
[DEVENGAGE-1429] --filtercondition flag for the CLI list command 

### PR DESCRIPTION
**Pull request by charliecon**

### CODE REVIEW

File: resources/sdk/clisdkclient/extensions/utils/filter.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+		allMatchedObjects []interface{}
+		currentObject     interface{}
+		match             bool
+		err               error
+	)
+
+	for _, object := range returnedObjects {
+		currentObject = object
+		for i, k := range keys {
+			if currentObjectMap, ok := currentObject.(map[string]interface{}); ok {
```
Comment (carnellj-genesys) [inline resources/sdk/clisdkclient/extensions/utils/filter.go:83]: Hi Charlie,

I would refactor the code below to get rid of the if..then..else.  Properly structured go will do a | line down from each if..then statement.  So you check something and grab the data and return rather then doing a nest of if..then statements.

File: resources/sdk/clisdkclient/extensions/utils/filter.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func fieldMatchesValue(jsonValue interface{}, cliInputValue string, operator string) (bool, error) {
+	if fieldValueStr, ok := jsonValue.(string); ok {
+		return compareStrings(fieldValueStr, cliInputValue, operator)
+	} else if fieldValueBool, ok := jsonValue.(bool); ok {
```
Comment (carnellj-genesys) [inline resources/sdk/clisdkclient/extensions/utils/filter.go:135]: See my earlier comments about if..then

**Review by carnellj-genesys — APPROVED** [review]
Just a couple of small items, but otherwise looks good

File: resources/sdk/clisdkclient/extensions/utils/filter.go (hunk 1 above)
Comment (charliecon) [reply resources/sdk/clisdkclient/extensions/utils/filter.go]: @carnellj-genesys  done ✅

**Pull request merged**